src/Container/Container.php

```diff
@@ -741,7 +741,12 @@ protected function resolveDependencies( array $dependencies ) {
 			// If the class is null, it means the dependency is a string or some other
 			// primitive type which we can not resolve since it is not a class and
 			// we will just bomb out with an error since we have no-where to go.
-			$result = is_null( $dependency->getClass() )
+			// Note: ReflectionParameter::getClass() is deprecated since PHP 8
```

Comment:
> Can we also add a @note that getType() isn't supported until php7? I'd like to make it clear why we have this if/then/else logic. At first glance it seems like a quick cleanup task to just use getType() instead of getClass(), but it isn't that simple.

Same for this usage.

Reply:
Resolved e8dc6e60